src/super_gradients/training/datasets/segmentation_datasets/segmentation_dataset.py

```diff
@@ -1,102 +1,55 @@
-            self.sample_loader = sample_loader
-        if target_loader is not None:
-            self.target_loader = target_loader
 
         # CREATE A DIRECTORY DATASET OR A LIST DATASET BASED ON THE list_file INPUT VARIABLE
         if list_file is not None:
             ListDataset.__init__(self, root=root, file=list_file, target_extension=target_extension,
-                                 sample_loader=self.sample_loader, sample_transform=self.sample_transform,
-                                 target_loader=self.target_loader, target_transform=self.target_transform,
+                                 sample_loader=self.sample_loader,
+                                 target_loader=self.target_loader,
                                  collate_fn=collate_fn)
         else:
             DirectoryDataSet.__init__(self, root=root, samples_sub_directory=samples_sub_directory,
                                       targets_sub_directory=targets_sub_directory, target_extension=target_extension,
-                                      sample_loader=self.sample_loader, sample_transform=self.sample_transform,
-                                      target_loader=self.target_loader, target_transform=self.target_transform,
+                                      sample_loader=self.sample_loader,
+                                      target_loader=self.target_loader,
                                       collate_fn=collate_fn)
-        # DEFAULT TRANSFORMS
-        # FIXME - Rescale before RandomRescale is kept for legacy support, consider removing it like most implementation
-        #  papers regimes.
-        default_image_mask_transforms_aug = transform.Compose([RandomFlip(),
-                                                               Rescale(short_size=self.img_size),
-                                                               RandomRescale(scales=(0.5, 2.0)),
-                                                               RandomRotate(),
-                                                               PadShortToCropSize(self.crop_size),
-                                                               CropImageAndMask(crop_size=self.crop_size,
-                                                                                mode="random"),
-                                                               RandomGaussianBlur()])
 
-        self.image_mask_transforms_aug = image_mask_transforms_aug or default_image_mask_transforms_aug
-        # FIXME: CROP SIZE CANNOT BE PASSED WHEN LIST
-        if image_mask_transforms is None:
-            image_mask_transforms = transform.Compose([Rescale(short_size=self.crop_size),
-                                                       CropImageAndMask(crop_size=self.crop_size, mode="center")
-                                                       ])
-
-        self.image_mask_transforms = image_mask_transforms
+        self.transforms = transform.Compose(transforms) if transforms else []
```

Comment:
[] won't be subscriptable...

Reply:
[] is passed to `Compose` so you can still do:
`transformed = self.transforms({"image": image, "mask": mask})`
Even if there are no transforms passed to the Dataset.
Otherwise you would need some kind of `if else` to handle it.
What is the scenario you are worried about?

Reply:
I don't see where it will be resolved. Note that the factory resolved the argument, not the attribute (I looked up self.transforms usages, and didn't see it being passed to compose after this line).

Reply:
Got it. Nice catch!